recipe-app/ViewModel/RecipeListViewModel.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  RecipeListViewModel.swift
+//  recipe-app
+//
+//  Created by Arthur Hermann on 27/11/2024.
+//
+
+import Foundation
+import UIKit
+
+ class RecipeListViewModel{
+    let key = "5b91aa819e6d4bc8848f4c972103e6dc"
+    var recipes = [Recipe]()
+    
+     /*
+      Retrieves first 20 recipes matching the user's search.
+      */
+     func searchRecipe(_ query: String) {
+         guard let url = URL(string: "https://api.spoonacular.com/recipes/complexSearch?apiKey=\(key)&query=\(query)&number=20") else {return}
+         
+         DispatchQueue.global(qos: .background).async { [weak self] in
+       
+             /*
+              Requests recipes from API and adds them to recipe list if the request is successful.
+              */
+         let session = URLSession.shared
+             session.dataTask(with: url) { data, response, error in
+                 if let response = response{
+                     if let data = data {
+                         if let httpResponse = response as? HTTPURLResponse {
+                             if httpResponse.statusCode == 200 {
+                                 self?.parse(json: data)
+                             } else {
+                                 print("Unexpected status code: \(httpResponse.statusCode)")
+                             }
+                         }
+                     }
+                 }
```

Comment:
`TLDR;` You need to .resume your task to be able to fetch your data. 

When you create a dataTask using URLSession, it returns a URLSessionDataTask object. This object represents a network task that is paused by default. To start the task, you must explicitly call `.resume()`.

This design gives you control over when the task begins, enabling tasks to be configured or queued before execution. For example:

* You might want to add headers or modify task priorities before starting it.
* In some cases, you may pause tasks and resume them later.